src/data/lex.rs

```diff
@@ -371,7 +371,19 @@ impl std::fmt::Display for Literal {
             Int(i) => write!(f, "{}", i),
             UnsignedInt(u) => write!(f, "{}", u),
             Float(n) => write!(f, "{}", n),
-            Str(s) => write!(f, "\"{}\"", String::from_utf8_lossy(s)),
+            Str(s) => {
+                let escaped = s
+                    .into_iter()
+                    .flat_map(|c| match c {
+                        b'\n' => "\\n".bytes().collect(),
+                        b'\r' => "\\r".bytes().collect(),
+                        b'\t' => "\\t".bytes().collect(),
```

Comment:
I think this would be better as
```suggestion
                        b'\n' => b"\\n",
                        b'\r' => b"\\r",
                        b'\t' => b"\\t",
```

Reply:
The problem is that `b""` returns a `&[u8; 2]` but `vec![*c]` has not the same type. We can also not use `&[*c]` or something similar because that would either has the wrong type, or would create a temporary value that is returned by the closure.

Reply:
Hmm, that makes sense, and `[*c]` wouldn't work because it's length 1 instead of 2. I'd be interested if `[*c].as_slice()` would work but I bet that runs into lifetime issues.